server/aesdsocket.c

```diff
@@ -177,150 +362,242 @@ int main(int argc, char *argv[])
+    	}
+
+		if(timer_settime(timerid, TIMER_ABSTIME, &interval, NULL ) != 0 ) 
+		{		   
+			syslog(LOG_ERR, "timer_settime failed");
+			exit_on_error = true;
+			goto EXITING;
+		} 
+    }
+    
+    while(exit_on_signal==0) 
+    {
+
+		len = sizeof(struct sockaddr);
+		
+		//accept connection
 		acceptedfd = accept(sockfd, (struct sockaddr *) &saddr, &len);
+		
+		if(exit_on_signal || exit_on_error)
+			break;
+			
 		if (acceptedfd == -1)
 		{
 			syslog(LOG_ERR, "socket accepting failed\n");
-			closeall();
-			exit(EXIT_FAILURE);
+			exit_on_error = true;
+			goto EXITING;
 		}
-
+		if(exit_on_signal || exit_on_error)
+			break;
+			
 		syslog(LOG_DEBUG, "Accepted connection from '%s'\n", inet_ntoa((struct in_addr)saddr.sin_addr));
-	
 
-		//Signals from set are added to invoking process's signal mask
-        if (sigprocmask(SIG_BLOCK, &mask, NULL)) 
+		datap = malloc(sizeof(slist_data_t));
+		if (datap == NULL)
 		{
-            syslog(LOG_ERR, "sigprocmask");
-			closeall();
-            exit(EXIT_FAILURE);
-        }
-
-		do
-		{
-			memset(buffer, 0, sizeof(buffer));
-			
-			nbytes = recv(acceptedfd, buffer, sizeof(buffer), 0);
-
-			if (nbytes)
-			{
-				// syslog(LOG_DEBUG, "nbytes - %d\n", nbytes);
-				// syslog(LOG_DEBUG, "'%s'\n", buffer);
-				if(nbytes > strlen(buffer))
-					nr = write(fd, buffer, strlen(buffer));
-				else
-					nr = write(fd, buffer, nbytes);
-					
-				if (nr == -1)	
-				{//if error
-					syslog(LOG_ERR, "can't write received string in file '%s'\n", filepath);
-					break;	
-				}
-				total_bytes += nr;
-			}
-			else
-				break;
-
-		}while(strchr(buffer, '\n') == NULL);
-		
-				
-		nbytes=0;
-		lseek(fd, 0, SEEK_SET);
-		
-		sendbuffer = (char *)malloc(sizeof(char) * (total_bytes+location));
-		if (sendbuffer == NULL)
 			syslog(LOG_ERR, "malloc failed\n");
-		
-		memset(sendbuffer, 0, total_bytes+location);
-		nr = read(fd, sendbuffer, total_bytes+location);
-		if (nr != total_bytes+location)	
-		{//if error
-			syslog(LOG_ERR, "can't read proper bytes from file '%s'\n", filepath);
-			break;	
+			exit_on_error = true;
+			goto EXITING;
 		}
-		
-		//syslog(LOG_DEBUG, "'%s'\n", sendbuffer);
-		nbytes = send(acceptedfd, sendbuffer, nr, 0);
-		if(nbytes != nr)
+			
+		SLIST_INSERT_HEAD(&head,datap,entries);
+		datap->params.acceptedfd = acceptedfd;
+		datap->params.complete_status_flag = false;
+		datap->params.thread_complete_success = false;
+		datap->params.sin_addr = saddr.sin_addr;
+		datap->params.fd=fd;
+
+		pthread_create(&(datap->params.thread), NULL, (void*)&packetRWthread, (void*)&(datap->params));
+
+		SLIST_FOREACH(datap,&head,entries)
 		{
-			syslog(LOG_ERR, "not all bytes sent\n");
-			break;
-		}
-		//syslog(LOG_DEBUG, "nbytes - %d\n", nbytes);
-		//total_bytes -= nbytes;
-		total_bytes = 0;
-		
-		location=lseek(fd, 0, SEEK_END);
+		    if (datap->params.complete_status_flag == true)
+		        pthread_join(datap->params.thread,NULL);
+		    else
+		    	continue;
+    	}
```

Comment:
@MuktaDarekar watch your indents and spacing here to make your code more readable.

Reply:
My text editor is giving me this problem, modified it using another editor and updated.